x/streamer/types/distr_info.go

```diff
@@ -32,3 +35,33 @@ func (r DistrRecord) ValidateBasic() error {
 	}
 	return nil
 }
+
+var hundred = math.NewInt(100)
+
+// DistrInfoFromDistribution converts sponsorship distribution to the DistrInfo type and performs a
+// basic validation for DistrInfo.Records. Returning an empty DistrInfo (with zero DistrInfo.TotalWeight)
+// is a valid scenario.
+func DistrInfoFromDistribution(d sponsorshiptypes.Distribution) (*DistrInfo, error) {
+	totalWeight := math.ZeroInt()
+	records := make([]DistrRecord, 0, len(d.Gauges))
+	for _, g := range d.Gauges {
+		// d.VotingPower is always > 0 to the Distribution type contract
+		weight := g.Power.Mul(hundred).Quo(d.VotingPower)
+
+		record := DistrRecord{
+			GaugeId: g.GaugeId,
+			Weight:  weight,
+		}
+		if err := record.ValidateBasic(); err != nil {
+			return nil, err
+		}
+
+		totalWeight = totalWeight.Add(weight)
```

Comment:
isn't totalWeight always 100 given sum(gagues.power) == VotingPower (which I assume is an invariant?)